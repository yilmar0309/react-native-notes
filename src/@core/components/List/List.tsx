import React from 'react';
import { Text, View, VirtualizedList } from 'react-native';
import { List as ImmutableList, Map as ImmutableMap } from 'immutable';

import { ConfigEntity } from '@hooks/useConfigTheme';

import useStyles from './styles';

interface Props {
  configTheme: ConfigEntity;
  dataSource: ImmutableList<ImmutableMap<string, any>>;
  renderItem: (item: any) => JSX.Element;
  refreshing: boolean;
  onEndReachedThreshold?: number;
  onEndReached?: () => void;
  onRefresh?: () => void;
  extraData: any;
}

const List: React.FC<Props> = (props) => {
  const { configTheme } = props;
  const styles = useStyles(configTheme);

  const { dataSource } = props;
  const getItem = (data: any, index: number) => data.get(index);
  const getItemCount = () =>
  dataSource.size;
  const keyExtractor = (item: ImmutableMap<string, any>, index: number) => `${item.size}-${index}`;
  const noFoundData = () => <Text style={styles.text}>No hay informaci√≥n *</Text>;

  return (
    <View style={styles.container}>
      <VirtualizedList
        { ...props }
        maxToRenderPerBatch={5}
        initialNumToRender={5}
        getItem={getItem}
        getItemCount={getItemCount}
        data={dataSource}
        keyExtractor={keyExtractor}
        ListEmptyComponent={noFoundData}
      />
    </View>
  )
}

export default List;
