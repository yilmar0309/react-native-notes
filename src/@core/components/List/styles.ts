import { StyleSheet } from 'react-native';
import { ConfigEntity } from '@hooks/useConfigTheme';

export default (props: ConfigEntity) => {
  return StyleSheet.create({
    container: {
      flex: 1,
    },
    text: {
      marginTop: 20,
      textAlign: 'center',
    },
  })
}
