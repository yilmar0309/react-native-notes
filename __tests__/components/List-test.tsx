/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 import { fromJS, List as ImmutableList } from 'immutable';
 // Note: test renderer must be required after react-native.
 import renderer from 'react-test-renderer';
 
 import List from '@components/List/List';

 const mockedNavigate = jest.fn();
 
 jest.mock('@react-navigation/native', () => {
   return {
     ...jest.requireActual('@react-navigation/native'),
     useNavigation: () => ({
       navigate: mockedNavigate,
     }),
   };
 });
 
 test('renders correctly', () => {
   const configTheme = {
     primary: '#FE4A01',
     secondary: '#FEE2CE',
     backgroundScreens: '#f2f2f4',
     textLight: 'white',
     textDark: 'black',
     textPrimary: '#000000',
     textSecondary: '#B9B9BB',
     textActionPrimary: '',
     textButtonPrimary: 'white',
     textButtonSecondary: '#1e344f',
     textInputTitle: '#B9B9BB',
     textInputText: 'black',
     card: 'white',
   }
   const renderItem = jest.fn();
   const onRefresh = jest.fn();
   const loadingRefresh = false;
   const data = ImmutableList([])
   const testRenderer = renderer.create(<List 
      dataSource={data}
      extraData={fromJS([])}
      renderItem={renderItem}
      refreshing={loadingRefresh}
      onRefresh={onRefresh}
      configTheme={configTheme} 
    />)
   const testInstance = testRenderer.root;

   expect(testInstance.findByType(List).props.foo);
 });
 
 