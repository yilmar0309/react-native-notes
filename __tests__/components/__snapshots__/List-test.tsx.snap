// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f2f2f4",
      "flex": 1,
    }
  }
>
  <Text>
    List
  </Text>
</View>
`;
